Core/Versioning/ModuleVersion.cs

```diff
@@ -54,7 +54,7 @@ public ModuleVersion(string version)
             // If we have an epoch, then record it.
             if (match.Groups["epoch"].Value.Length > 0)
             {
-                _epoch = Convert.ToInt32( match.Groups["epoch"].Value );
+                _epoch = Convert.ToInt64( match.Groups["epoch"].Value );
```

Comment:
`_epoch` is still just an int, though? Wouldn't this still overflow, just in the assignment to `int` rather than inside the `ToInt32/64` call?

```csharp
        private readonly int _epoch;
```